src/main/java/com/example/bank/bankTransfer/transfer/history/export/XlsxTransferHistoryGenerator.java

```diff
@@ -0,0 +1,159 @@
+package com.example.bank.bankTransfer.transfer.history.export;
+
+import com.example.bank.bankTransfer.account.Account;
+import com.example.bank.bankTransfer.account.AccountRepository;
+import com.example.bank.bankTransfer.transfer.history.TransferHistory;
+import com.example.bank.bankTransfer.transfer.history.TransferHistoryService;
+import lombok.RequiredArgsConstructor;
+import org.apache.poi.ss.usermodel.*;
+import org.apache.poi.xssf.usermodel.XSSFWorkbook;
+import org.jetbrains.annotations.NotNull;
+import org.springframework.stereotype.Service;
+
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.List;
+
+@Service
+@RequiredArgsConstructor
+public class XlsxTransferHistoryGenerator {
```

Comment:
Przykladowy refactoring : 
`@Service
@RequiredArgsConstructor
public class XlsxTransferHistoryGenerator {

    private static final String TRANSFER_HISTORY_SHEET_NAME = "Transfer history";
    private static final String FILE_NAME_PATTERN = "Transfer history client_%d.xlsx";
    private static final int DATE_COLUMN_WIDTH = 4900;
    private static final int HORIZONTAL_PADDING = 200;
    private static final List<String> HEADER_CELL_TITLES = List.of(
        "No.", "Created on", "Client ID", "Transfer type", "Bank account number", 
        "Title of transfer", "Amount", "Balance"
    );
    
    private final AccountRepository accountRepository;
    private final TransferHistoryService transferHistoryService;

    public void generateXlsxTransferHistory(Long clientId) {
        try (Workbook workbook = new XSSFWorkbook()) {
            initializeWorkbook(workbook, clientId);
        } catch (IOException ex) {
            throw new RuntimeException("Error generating transfer history XLSX", ex);
        }
    }

    private void initializeWorkbook(Workbook workbook, Long clientId) {
        Sheet sheet = workbook.createSheet(TRANSFER_HISTORY_SHEET_NAME);
        createHeaderRow(sheet);
        int numberOfTransferHistories = transferHistoryService.transferHistoryForClient(clientId).size();
        createDataRows(sheet, numberOfTransferHistories, clientId);
        autoSizeColumns(sheet, HEADER_CELL_TITLES.size());
        saveWorkbookToFile(workbook, clientId);
    }

    private void createHeaderRow(Sheet sheet) {
        Row headerRow = sheet.createRow(0);
        CellStyle headerStyle = createStyleForHeader(sheet.getWorkbook());
        IntStream.range(0, HEADER_CELL_TITLES.size()).forEach(columnIndex -> {
            Cell cell = headerRow.createCell(columnIndex);
            cell.setCellValue(HEADER_CELL_TITLES.get(columnIndex));
            cell.setCellStyle(headerStyle);
        });
    }

    private void createDataRows(Sheet sheet, int rowCount, Long clientId) {
        CellStyle dateTimeCellStyle = createDateTimeCellStyle(sheet.getWorkbook());
        CellStyle decimalCellStyle = createDecimalCellStyle(sheet.getWorkbook());
        CellStyle defaultWrapTextStyle = createDefaultWrapTextStyle(sheet.getWorkbook());
        List<TransferHistory> transferHistories = transferHistoryService.transferHistoryForClient(clientId);

        for (int i = 0; i < rowCount; i++) {
            Row row = sheet.createRow(i + 1);
            TransferHistory history = transferHistories.get(i);
            Account account = accountRepository.findById(history.getBankAccountId())
                                 .orElseThrow(() -> new IllegalStateException("Account not found"));
            fillRowWithData(row, history, account, dateTimeCellStyle, decimalCellStyle, defaultWrapTextStyle);
        }
    }

    private void fillRowWithData(Row row, TransferHistory history, Account account, 
                                 CellStyle dateTimeCellStyle, CellStyle decimalCellStyle, 
                                 CellStyle defaultWrapTextStyle) {
        // tu implementacja wypelniania danymi 
    }

    // Tu te metody co styluja te wszystkie rzeczy 

    private void saveWorkbookToFile(Workbook workbook, Long clientId) {
        // zapis do pliku
    }
}
`

Reply:
A no i nawet tu mozna ulepszyc to co wstawilem bo pobieram sheet.getWorkbook kilka razy np w createDataRows a to zbedny syf ale pisane bylo na szybko :)